Sources/AppStoreConnectClient/AppStoreConnectClient.swift

```diff
@@ -60,40 +60,92 @@ public struct AppStoreConnectClient {
-            throw AppStoreConnectError.badRequest
+            switch result.body {
+            case .json(let json):
+                throw AppStoreConnectError.badRequest(errors: handleError(from: json))
+            }
         case .forbidden(let result):
-            throw AppStoreConnectError.forbidden
+            switch result.body {
+            case .json(let json):
+                throw AppStoreConnectError.forbidden(errors: handleError(from: json))
+            }
         case .unauthorized(let result):
-            throw AppStoreConnectError.unauthorized
+            switch result.body {
+            case .json(let json):
+                throw AppStoreConnectError.unauthorized(errors: handleError(from: json))
+            }
         case .undocumented(let statusCode, _):
             throw AppStoreConnectError.serverError(errorCode: statusCode)
         }
     }
     
+    /// Converts an error response into a formatted string.
+    /// - Parameter response: The error response object.
+    /// - Returns: A formatted string describing the error response, or `nil` if the response is empty.
+    private func handleError(from response: Components.Schemas.ErrorResponse) -> String? {
+        var stringError = ""
+        response.errors?.forEach { error in
+            stringError.append("""
+        \nThe request failed with: \(error.status), \(error.code), \(error.title), \(error.detail).
+        """)
+        }
+        guard !stringError.isEmpty else {
+            return nil
+        }
+        return stringError
+    }
+    
     /// Fetches a collection of versions for a specified app from the App Store Connect API.
     /// - Parameter app: The app for which to fetch versions.
     /// - Returns: An array of `Release` objects.
-    /// - Throws: An error if the fetch operation fails.
+    /// - Throws: An error of type `AppStoreConnectError` if the fetch operation fails.
     public func fetchVersions(for app: Application) async throws -> [Release] {
         let response = try await client.apps_hyphen_appStoreVersions_hyphen_get_to_many_related(
             path: .init(id: app.id)
         )
         switch response {
         case .ok(let okResponse):
-            switch okResponse.body {
-            case .json(let json):
+            if case .json(let json) = okResponse.body {
                 return json.data.compactMap({ Release(schema: $0) })
             }
+        default:
+            return try await errorResponse(from: response)
+        }
+        throw AppStoreConnectError.unexpectedError(errors: "\(response)")
+    }
+    
+    /// Handles error responses returned by the App Store Connect API.
+    /// - Parameter response: The response received from the API.
+    /// - Throws: An error of type `AppStoreConnectError` if the response indicates an error.
+    func errorResponse(
+        from response: Operations.apps_hyphen_appStoreVersions_hyphen_get_to_many_related.Output
+    ) async throws -> [Release] {
```

Comment:
Do not see any reason why this function should be async

Reply:
Changed.